### PR TITLE
Passing Active Editor Info 

### PR DESCRIPTION
**Pull request by amitjoshi438**
This pull request includes various updates to enhance functionality, improve code structure, and support new features. The changes focus on adding new API proposals, modifying constructors and methods for better context handling, and integrating new utility functions.

### Enhancements and New Features:
* **`package.json`**: Added support for new API proposals including `chatParticipant`, `chatVariableResolver`, and `languageModels`.
* **`PowerPagesChatParticipant.ts`**: Introduced a new utility function `createAndReferenceLocation` for referencing code locations in chat responses. [[1]](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29L142-R175) [[2]](diffhunk://#diff-b379cc33401ef3e60e20ce6fd0a4fa00a14fd527982f762b7d025a0f576c603fR60-R70)

### Code Structure Improvements:
* **`OrgChangeNotifier.ts`**: Updated the constructor and `createOrgChangeNotifierInstance` method to include `vscode.ExtensionContext` for better context management.
* **`PowerPagesCopilot.ts`**: Refactored to include active file content and parameters in API requests, improving the relevance of responses. [[1]](diffhunk://#diff-98c063ddf20a69c9594332e9f968fe00b2e009c5de85256f9e336c089e6880abL239-R239) [[2]](diffhunk://#diff-98c063ddf20a69c9594332e9f968fe00b2e009c5de85256f9e336c089e6880abR370-R371)

### Utility and Configuration Updates:
* **`IntelligenceApiService.ts`**: Added support for testing with localhost by configuring an HTTPS agent and setting appropriate headers.
* **`model.ts`**: Expanded `IActiveFileData` interface to include `activeFileUri`, `startLine`, and `endLine` for detailed file context.

These changes collectively enhance the functionality and maintainability of the codebase, ensuring better integration and context handling across different components.

### CODE REVIEW

File: src/common/copilot/PowerPagesCopilot.ts [hunk 1]
```diff
@@ -385,7 +386,7 @@ export class PowerPagesCopilot implements vscode.WebviewViewProvider {
                     }
 
                 }
-                return sendApiRequest(data, activeFileParams, orgID, intelligenceApiToken, sessionID, metadataInfo.entityName, componentInfo, telemetry, this.aibEndpoint, this.geoName, this.crossGeoDataMovementEnabledPPACFlag);
+                return sendApiRequest([{ displayText: data[0].displayText, code: activeFileContent }], activeFileParams, orgID, intelligenceApiToken, sessionID, metadataInfo.entityName, componentInfo, telemetry, this.aibEndpoint, this.geoName, this.crossGeoDataMovementEnabledPPACFlag);
```
Comment (gshivi) [inline src/common/copilot/PowerPagesCopilot.ts:389]: Will data[0] always exist? Do we want to add null check here?

**Review by gshivi — APPROVED** [review]

File: src/common/copilot/PowerPagesCopilot.ts (hunk 1 above)
Comment (amitjoshi438) [reply src/common/copilot/PowerPagesCopilot.ts:389]: Yes, it's user prompt and will always exist.

File: src/client/OrgChangeNotifier.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export class OrgChangeNotifier {
         if (this._pacWrapper) {
             this.setupFileWatcher();
         }
+
```
Comment (tyaginidhi) [inline src/client/OrgChangeNotifier.ts:28]: stray change

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts [hunk 3]
```diff
@@ -57,6 +57,17 @@ export function isEntityInSupportedList(entity: string): boolean {
     return SUPPORTED_ENTITIES.includes(entity);
 }
 
+export function createAndReferenceLocation(activeFileUri: vscode.Uri, startLine: number, endLine: number): vscode.Location {
+
+    const positionStart = new vscode.Position(startLine, 0);
+    const positionEnd = new vscode.Position(endLine, 0);
+
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts:64]: nit: we can club all these variables together in definition at start of function

**Review by tyaginidhi — APPROVED** [review]

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts (hunk 3 above)
Comment (amitjoshi438) [reply src/common/chat-participants/powerpages/PowerPagesChatParticipantUtils.ts]: Thanks, done.

**Pull request merged**